Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/DefaultRiskLevels.kt

```diff
@@ -216,16 +159,313 @@ class DefaultRiskLevels @Inject constructor() : RiskLevels {
             Timber.d("Risk level changed and notification sent. Current Risk level is ${riskLevel.raw}")
         }
         if (lastCalculatedScore.raw == RiskLevelConstants.INCREASED_RISK &&
-            riskLevel.raw == RiskLevelConstants.LOW_LEVEL_RISK) {
+            riskLevel.raw == RiskLevelConstants.LOW_LEVEL_RISK
+        ) {
             LocalData.isUserToBeNotifiedOfLoweredRiskLevel = true
 
             Timber.d("Risk level changed LocalData is updated. Current Risk level is ${riskLevel.raw}")
         }
         RiskLevelRepository.setRiskLevelScore(riskLevel)
     }
 
+    private fun dropDueToMinutesAtAttenuation(
+        exposureWindow: ExposureWindow,
+        attenuationFilters: List<RiskCalculationParametersOuterClass.MinutesAtAttenuationFilter>
+    ) =
+        attenuationFilters.any { attenuationFilter ->
+            // Get total seconds at attenuation in exposure window
+            val secondsAtAttenuation = exposureWindow.scanInstances
+                .filter { attenuationFilter.attenuationRange.inRange(it.typicalAttenuationDb) }
+                .fold(0) { acc, scanInstance -> acc + scanInstance.secondsSinceLastScan }
+
+            val minutesAtAttenuation = secondsAtAttenuation / 60
+            return attenuationFilter.dropIfMinutesInRange.inRange(minutesAtAttenuation)
+        }
+
+    private fun determineTransmissionRiskLevel(
```

Comment:
We could make all these functions working on `ExposureWindow` extension functions on it.

Reply:
done in #1656